### PR TITLE
lint: add new errwrap linter

### LINKED ISSUE #42510: sql: flattening error objects into other errors erases telemetry data and user hints
I found this while looking for something else:

```go
 func NewInvalidSchemaDefinitionError(err error) error {
       return pgerror.New(pgcode.InvalidSchemaDefinition, err.Error())
 }
```

By flattening the error with `err.Error()` this constructor strips the error object of all its details, including stack traces but also **user-directed hints, links to documentation issues and telemetry data**.

Other examples (non exhaustive):
```
execute.go:                     return nil, pgerror.New(pgcode.WrongObjectType, err.Error())
export.go:                      return nil, pgerror.New(pgcode.InvalidParameterValue, err.Error())
rowexec/hashjoiner.go:                                  err = pgerror.Wrapf(addErr, pgcode.OutOfMemory, "while spilling: %v", err)
set_var.go:             return wrapSetVarError("statement_timeout", s, "%v", err)
```

There are multiple ways to fix it:

- prefer `errors.Wrap[f]` when no need to add a pg code
- use `pgerror.WithCandidateCode` when need to add a pg code if there wasn't one already to start with
- use `errors.Combine` or `errors.WithSecondaryError` to combine errors together

I am fixing the first example at the top in #42509, but the other cases remain to be fixed.

cc @cockroachdb/sql-execution

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/42510

This linter checks if we don't correctly wrap errors.

The `/* nolint:errwrap */` comment can be used to disable the linter inline.

See individual commits for mistakes this linter caught.
It had already caught a few in https://github.com/cockroachdb/cockroach/pull/72353, https://github.com/cockroachdb/cockroach/pull/72352, https://github.com/cockroachdb/cockroach/pull/72351, https://github.com/cockroachdb/cockroach/pull/72350, and https://github.com/cockroachdb/cockroach/pull/72349.

Release note: None

### CODE REVIEW

**Review by knz — COMMENTED** [review]
tests LGTM but you forgot a few cases.

> Reviewed 16 of 16 files at r1, 13 of 13 files at r2, 11 of 11 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-:-MqjXXOy0NnKRvoClyxI:b3wob3b)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[pkg/server/drain_test.go, line 90 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqjW-lH8VaEzoSfNsQE:-MqjW-lI9OIYGYmafoVp:b-lc5q36) ([raw file](https://github.com/cockroachdb/cockroach/blob/c498e14597a27e552407799a527846d1bbc99bb6/pkg/server/drain_test.go#L90)):*
> ```Go
> 			return nil
> 		}
> 		return errors.Newf("server not yet refusing RPC, got %v", err /* nolint:errwrap */)
> ```

Can you explain in a comment why the linter exception is needed here?
___
*[pkg/testutils/lint/passes/errwrap/errwrap.go, line 104 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqjWroaBLfxUl_BnR3f:-MqjWroaBLfxUl_BnR3g:b-pu30zc) ([raw file](https://github.com/cockroachdb/cockroach/blob/573da3a3881ff3d4c7b130781970b995b4454715/pkg/testutils/lint/passes/errwrap/errwrap.go#L104)):*
> ```Go
> 
> 		// Check that none of the arguments are err.Error()
> 		if (pkg.Name() == "fmt" && fn.Name() == "Errorf") ||
> ```

Can you implement this via a map? This would be more efficient.

Also is there a way to share this data with the map in `../fmtsafe/functions.go`? 
(This also reveals you're missing a few here - unimplemented, roachpb, sqlerrors etc)

**Comment by postamar** [comment]
Thanks for doing this! This is valuable.

**Comment by andreimatei** [comment]
Just in case it's not in there already, please make sure there's a way to inhibit the linter manually at code sites.

**Review by rickystewart — CHANGES_REQUESTED** [review]
small request, can you please make the appropriate change for bazel too?

```
diff --git a/BUILD.bazel b/BUILD.bazel
index f1b5c20697..4e322ae951 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -238,6 +238,7 @@ nogo(
         "//pkg/testutils/lint/passes/descriptormarshal",
         "//pkg/testutils/lint/passes/errcheck",
         "//pkg/testutils/lint/passes/errcmp",
+        "//pkg/testutils/lint/passes/errwrap",
         "//pkg/testutils/lint/passes/fmtsafe",
         "//pkg/testutils/lint/passes/grpcclientconnclose",
         "//pkg/testutils/lint/passes/grpcstatuswithdetails",

```

**Review by rafiss — COMMENTED** [review]
updated bazel build config

i'm giving this another run now that more error functions are being linted

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-:-MqpMk1T2wznBCRBpKt5:b2s5f2v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @otan)

___
*[pkg/server/drain_test.go, line 90 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqjW-lH8VaEzoSfNsQE:-MqpEhJ0-O-48V6Y5vLV:b5j26t2) ([raw file](https://github.com/cockroachdb/cockroach/blob/c498e14597a27e552407799a527846d1bbc99bb6/pkg/server/drain_test.go#L90)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Can you explain in a comment why the linter exception is needed here?
</blockquote></details>

yes added; it would be incorrect to wrap the error here since this function should return an error if `err` is nil, and wrapping a `nil` error results in another `nil` error.
___
*[pkg/testutils/lint/passes/errwrap/errwrap.go, line 104 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqjWroaBLfxUl_BnR3f:-MqpMecI-mQPixe8Wbf1:bhbicd9) ([raw file](https://github.com/cockroachdb/cockroach/blob/573da3a3881ff3d4c7b130781970b995b4454715/pkg/testutils/lint/passes/errwrap/errwrap.go#L104)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Can you implement this via a map? This would be more efficient.

Also is there a way to share this data with the map in `../fmtsafe/functions.go`? 
(This also reveals you're missing a few here - unimplemented, roachpb, sqlerrors etc)
</blockquote></details>

i made the maps backed by the same thing

File: pkg/testutils/lint/passes/errwrap/errwrap.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+			(pkg.Name() == "errors" && fn.Name() == "AssertionFailedWithDepthf") ||
+			(pkg.Name() == "errors" && fn.Name() == "NewAssertionErrorWithWrappedErrf") {
+			formatStringIdx = 1
+		} else if (pkg.Name() == "pgerror" && fn.Name() == "Wrapf") ||
+			(pkg.Name() == "pgerror" && fn.Name() == "NewWithDepthf") ||
+			(pkg.Name() == "errors" && fn.Name() == "WrapWithDepthf") {
+			formatStringIdx = 2
+		} else if pkg.Name() == "pgerror" && fn.Name() == "WrapWithDepthf" {
+			formatStringIdx = 3
+		} else {
```
[the remark this answers is not in the document]
Comment (rafiss) [reply pkg/testutils/lint/passes/errwrap/errwrap.go:153]: agreed, that sounds better. that's what i've done based on @knz's comment as well

**Review by ajwerner — APPROVED** [review]
:lgtm: mod the one testing nit. 

> Reviewed 1 of 12 files at r4, 10 of 17 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-:-MqpaEx89geTu1aGoc01:b-m4uswr)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @knz, @otan, @rafiss, and @rickystewart)

___
*[pkg/testutils/lint/passes/errwrap/testdata/src/a/a_test.go, line 63 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqpa7PxDiXSCg_ialqR:-Mqpa7PxDiXSCg_ialqS:b-ofdfnd) ([raw file](https://github.com/cockroachdb/cockroach/blob/205e71ef12866e144dae757ea0c748dcdc521467/pkg/testutils/lint/passes/errwrap/testdata/src/a/a_test.go#L63)):*
> ```Go
> 	_ = errors.NewAssertionErrorWithWrappedErrf(anotherErr, "format %v", wrappedErr) // want `non-wrapped error is passed to errors.NewAssertionErrorWithWrappedErrf; use pgerror.Wrap/errors.Wrap/errors.CombineErrors/errors.WithSecondaryError/errors.NewAssertionErrorWithWrappedErrf instead`
> 
> 	_ = fmt.Errorf("got %s", wrappedErr /* nolint:errwrap */)  // linting is skipped.
> ```

can you test the `//` style comment too? It should work.

**Comment by rickystewart** [comment]
The Bazel build surfaced an error, I assume this is legitimate?

```
21:00:58     compilepkg: nogo: errors found by nogo during build-time code analysis:
21:00:58     /tmp/rules_go_work-3000709617/github.com/cockroachdb/cockroach/pkg/util/httputil/http.go:116:16: non-wrapped error is passed to errors.Errorf; use pgerror.Wrap/errors.Wrap/errors.CombineErrors/errors.WithSecondaryError/errors.NewAssertionErrorWithWrappedErrf instead (errwrap)
```

**Review by knz — COMMENTED** [review]
Nice, a few nits and this is good to go.

> Reviewed 8 of 38 files at r6, 40 of 40 files at r7, 9 of 9 files at r8, 17 of 17 files at r9, 17 of 17 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-:-Mqy-yUBAPGxaeMfS2z1:b1bx066)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @adityamaru, @ajwerner, @otan, @rafiss, and @rickystewart)

___
*[pkg/ccl/importccl/read_import_mysql.go, line 767 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxwvQcBgzfeWbljxEd:-MqxwvQd9HcIk5e0nvD4:bmi8ppe) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/ccl/importccl/read_import_mysql.go#L767)):*
> ```Go
> 					// error.
> 					// nolint:errwrap
> 					return nil, errors.Wrapf(
> ```

:100: 
___
*[pkg/cli/clierrorplus/decorate_error.go, line 90 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqxx9rw95Lojq2ycJGD:-Mqxx9rw95Lojq2ycJGE:b-wxvmuy) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cli/clierrorplus/decorate_error.go#L90)):*
> ```Go
> 
> 		connInsecureHint := func() error {
> 			const format = "cannot establish secure connection to insecure server.\n" +
> ```

hmm :thinking: 
is this a legitimate change? This suggests either it would benefit from errors.Wrap, or from a linter exception.

ditto below
___
*[pkg/cloud/amazon/s3_storage.go, line 437 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxRJRC8_dyC9oZImm:-MqxxRJS-7VnmTS27Yry:b-141qam) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/amazon/s3_storage.go#L437)):*
> ```Go
> 			case s3.ErrCodeNoSuchBucket, s3.ErrCodeNoSuchKey:
> 				// nolint:errwrap
> 				return nil, errors.WithMessage(
> ```

maybe `errors.Wrapf(errors.Wrap(...), "%v", err)` to ensure `err` gets embarked as a secondary error.
___
*[pkg/cloud/azure/azure_storage.go, line 160 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxXrc6uj2G16ccWz8:-MqxxXrc6uj2G16ccWz9:bmipv7j) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/azure/azure_storage.go#L160)):*
> ```Go
> 			case azblob.ServiceCodeBlobNotFound, azblob.ServiceCodeResourceNotFound:
> 				// nolint:errwrap
> 				return nil, 0, errors.WithMessage(
> ```

ditto previous, errors.Wrapf
___
*[pkg/cloud/gcp/gcs_storage.go, line 200 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxZij3RjwRBIGvSmP:-MqxxZij3RjwRBIGvSmQ:bmipv7j) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/gcp/gcs_storage.go#L200)):*
> ```Go
> 			// return our internal ErrFileDoesNotExist.
> 			// nolint:errwrap
> 			err = errors.WithMessage(
> ```

ditto previous, errors.Wrapf
___
*[pkg/cloud/httpsink/http_storage.go, line 270 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxaaH0nsaeqUQjLus:-MqxxaaH0nsaeqUQjLut:bmipv7j) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/httpsink/http_storage.go#L270)):*
> ```Go
> 		if err != nil && resp.StatusCode == 404 {
> 			// nolint:errwrap
> 			err = errors.WithMessage(
> ```

ditto previous, errors.Wrapf
___
*[pkg/cloud/nodelocal/nodelocal_storage.go, line 146 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxeZH-LCBssnX_9Zh:-MqxxeZH-LCBssnX_9Zi:b9he395) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/nodelocal/nodelocal_storage.go#L146)):*
> ```Go
> 		// The local store returns a golang native ErrNotFound, whereas the remote
> 		// store returns a gRPC native NotFound error.
> 		// nolint:errwrap
> ```

I don't think this linter exception is on the right source code line
___
*[pkg/cloud/nodelocal/nodelocal_storage.go, line 148 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxcNw35eb6X1_M-QK:-MqxxcNw35eb6X1_M-QL:bmipv7j) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/nodelocal/nodelocal_storage.go#L148)):*
> ```Go
> 		// nolint:errwrap
> 		if oserror.IsNotExist(err) || status.Code(err) == codes.NotFound {
> 			return nil, 0, errors.WithMessage(
> ```

ditto previous, errors.Wrapf
___
*[pkg/cmd/roachtest/tests/encryption.go, line 56 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxwFB6InciA5NNaaE:-MqxxwFB6InciA5NNaaF:b-l7w56d) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cmd/roachtest/tests/encryption.go#L56)):*
> ```Go
> 			// Generate encryption store key through `./cockroach gen encryption-key -s=size aes-size.key`.
> 			if err := c.RunE(ctx, c.Node(nodes), fmt.Sprintf("./cockroach gen encryption-key -s=%[1]d aes-%[1]d.key", size)); err != nil {
> 				return errors.Wrapf(err, "failed to generate aes key with size %d through CLI", size)
> ```

nit: aes -> AES
___
*[pkg/kv/kvserver/client_lease_test.go, line 669 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxyAbG51xNjVD49fd2:-MqxyAbG51xNjVD49fd3:bl6nlyl) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/kv/kvserver/client_lease_test.go#L669)):*
> ```Go
> 				wait(timetoDie.Nanoseconds())
> 			}
> 			// nolint:errwrap
> ```

nit: you can add the same comment `// NB: errors.Wrapf(nil, ...) returns nil.` that you added elsewhere.
___
*[pkg/kv/kvserver/replica_rangefeed_test.go, line 675 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxyVfX0JE7nEkYz2yg:-MqxyVfY5bzIbYgvqtkL:bpwwd7h) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/kv/kvserver/replica_rangefeed_test.go#L675)):*
> ```Go
> 			}
> 			err = repl.AdminTransferLease(ctx, roachpb.StoreID(1))
> 			return errors.Wrapf(err, "not raft follower: %+v, transferred lease", raftStatus)
> ```

beware, err can be nil here. I think you want the other thing instead.
___
*[pkg/security/securitytest/securitytest.go, line 80 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxzXDM2uc50l9QU4d9:-MqxzXDM2uc50l9QU4dA:b-8p4y34) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/security/securitytest/securitytest.go#L80)):*
> ```Go
> 			if _, dirErr := AssetDir(joined); dirErr != nil {
> 				errString := dirErr.Error()
> 				return nil, errors.Wrapf(err, "missing directory (%s)", errString)
> ```

you could take the dirErr as secondary error here:

```go
return nil, errors.Wrapf(err, "missing directory (%v)", dirErr /* nolint */)
```
___
*[pkg/util/httputil/http.go, line 116 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqy-nw89g0CKowNQVfY:-Mqy-nw89g0CKowNQVfZ:b4t8lg6) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/util/httputil/http.go#L116)):*
> ```Go
> 	if contentType := resp.Header.Get(ContentTypeHeader); !(resp.StatusCode == http.StatusOK && contentType == JSONContentType) {
> 		b, err := ioutil.ReadAll(resp.Body)
> 		// nolint:errwrap
> ```

nit: you can add the comment that explains what happens with nil errors here too

**Review by rafiss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71877)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @adityamaru, @ajwerner, @knz, @otan, @rafiss, and @rickystewart)

___
*[pkg/cli/clierrorplus/decorate_error.go, line 90 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqxx9rw95Lojq2ycJGD:-MqyzV9iBaWmrPGJHOoo:b-rahmth) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cli/clierrorplus/decorate_error.go#L90)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

hmm :thinking: 
is this a legitimate change? This suggests either it would benefit from errors.Wrap, or from a linter exception.

ditto below
</blockquote></details>

i made this change in order to match more closely to how the other error decorating functions worked in this file. (e.g. see `connSecurityHint` above.) i didn't use `errors.Wrap` here since it seemed like these messages were carefully formatted by hand, and `Wrap` makes it less obvious what the resulting string will look like. however, if your preference is for `Wrap` also, then i'd definitely agree with that change
___
*[pkg/cloud/amazon/s3_storage.go, line 437 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxRJRC8_dyC9oZImm:-Mqz0g495NyJ2bteywuj:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/amazon/s3_storage.go#L437)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

maybe `errors.Wrapf(errors.Wrap(...), "%v", err)` to ensure `err` gets embarked as a secondary error.
</blockquote></details>

done
___
*[pkg/cloud/azure/azure_storage.go, line 160 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxXrc6uj2G16ccWz8:-Mqz1zhJEr5dh-4-FcCj:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/azure/azure_storage.go#L160)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto previous, errors.Wrapf
</blockquote></details>

done
___
*[pkg/cloud/gcp/gcs_storage.go, line 200 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxZij3RjwRBIGvSmP:-Mqz2-NCAfZDYFJbJ0oF:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/gcp/gcs_storage.go#L200)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto previous, errors.Wrapf
</blockquote></details>

done
___
*[pkg/cloud/httpsink/http_storage.go, line 270 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxaaH0nsaeqUQjLus:-Mqz203QDaysbty62RMs:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/httpsink/http_storage.go#L270)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto previous, errors.Wrapf
</blockquote></details>

done
___
*[pkg/cloud/nodelocal/nodelocal_storage.go, line 146 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxeZH-LCBssnX_9Zh:-Mqz21-w5SsmLnnl-mxW:b-7ql1td) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/nodelocal/nodelocal_storage.go#L146)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I don't think this linter exception is on the right source code line
</blockquote></details>

fixed
___
*[pkg/cloud/nodelocal/nodelocal_storage.go, line 148 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxcNw35eb6X1_M-QK:-Mqz21rC-TCG34lp2kf1:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cloud/nodelocal/nodelocal_storage.go#L148)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto previous, errors.Wrapf
</blockquote></details>

done
___
*[pkg/cmd/roachtest/tests/encryption.go, line 56 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxxwFB6InciA5NNaaE:-Mqz25wT5RJh6LPQoENW:b-7ql1td) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cmd/roachtest/tests/encryption.go#L56)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

nit: aes -> AES
</blockquote></details>

fixed
___
*[pkg/kv/kvserver/client_lease_test.go, line 669 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxyAbG51xNjVD49fd2:-Mqz2DpJ4AqeAH3Fbyde:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/kv/kvserver/client_lease_test.go#L669)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

nit: you can add the same comment `// NB: errors.Wrapf(nil, ...) returns nil.` that you added elsewhere.
</blockquote></details>

done
___
*[pkg/kv/kvserver/replica_rangefeed_test.go, line 675 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxyVfX0JE7nEkYz2yg:-Mqz2G-WDpPAeKapIc0e:b-qtosaa) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/kv/kvserver/replica_rangefeed_test.go#L675)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

beware, err can be nil here. I think you want the other thing instead.
</blockquote></details>

fixed; very nice catch
___
*[pkg/security/securitytest/securitytest.go, line 80 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-MqxzXDM2uc50l9QU4d9:-Mqz2sYMAPLz22Ggwj6A:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/security/securitytest/securitytest.go#L80)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

you could take the dirErr as secondary error here:

```go
return nil, errors.Wrapf(err, "missing directory (%v)", dirErr /* nolint */)
```
</blockquote></details>

done
___
*[pkg/testutils/lint/passes/errwrap/testdata/src/a/a_test.go, line 63 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqpa7PxDiXSCg_ialqR:-Mqz36dO0vpQCoh_ZVvF:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/205e71ef12866e144dae757ea0c748dcdc521467/pkg/testutils/lint/passes/errwrap/testdata/src/a/a_test.go#L63)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

can you test the `//` style comment too? It should work. 
</blockquote></details>

done
___
*[pkg/util/httputil/http.go, line 116 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqy-nw89g0CKowNQVfY:-Mqz3JQIC6uHWzi0u1E5:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/util/httputil/http.go#L116)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

nit: you can add the comment that explains what happens with nil errors here too
</blockquote></details>

done

**Review by knz — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71877)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @adityamaru, @ajwerner, @knz, @otan, @rafiss, and @rickystewart)

___
*[pkg/cli/clierrorplus/decorate_error.go, line 90 at r10](https://reviewable.io/reviews/cockroachdb/cockroach/71877#-Mqxx9rw95Lojq2ycJGD:-Mqz5ofcAnrJTXa3YN7U:b-gmcet9) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb2175023e5003d2e967bc5b4c0fd9a759721b1b/pkg/cli/clierrorplus/decorate_error.go#L90)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

i made this change in order to match more closely to how the other error decorating functions worked in this file. (e.g. see `connSecurityHint` above.) i didn't use `errors.Wrap` here since it seemed like these messages were carefully formatted by hand, and `Wrap` makes it less obvious what the resulting string will look like. however, if your preference is for `Wrap` also, then i'd definitely agree with that change
</blockquote></details>

I don't have a strong opinion. If you keep your choice, you can explain it in a comment.

**Comment by rafiss** [comment]
thanks for your reviews!

bors r=ajwerner,knz

**Comment by rafiss** [comment]
bors r=ajwerner,knz

**Pull request merged**